Content.Server/NodeContainer/Nodes/AdjacentNode.cs

```diff
@@ -13,14 +13,17 @@ public override IEnumerable<Node> GetReachableNodes(TransformComponent xform,
             EntityQuery<NodeContainerComponent> nodeQuery,
             EntityQuery<TransformComponent> xformQuery,
             MapGridComponent? grid,
-            IEntityManager entMan)
+            IEntityManager entMan,
+            SharedMapSystem mapSystem)
         {
-            if (!xform.Anchored || grid == null)
+            if (!xform.Anchored
+                || xform.GridUid == null
+                || grid == null)
                 yield break;
```

Comment:
Kinda suss it's taking in the xform's griduid and the grid, just use Entity<Grid> for it and have the callers adjust.